File: python/day_1/day_1_1.py
import re

# read data
with open('./input.txt', "r") as f:
    data = f.read()

data = data.lower()

# remove trailing line break
if data[-1]=="\n":
    data = data[:-1]

# remove all non digit characters
cleaned = re.sub(r"[^\d\n]", "", data) 

# extract first and last digit of each string
values = [entry[0] + entry[-1]  for entry in cleaned.split("\n")]

# compute the sum of values
solution = sum([int(v) for v in values])

print('Solution:', solution)

File: julia/day_1/day_1_1.jl
sum = 0
f = open("input.txt", "r")
while !eof(f)  
    global sum
    s = readline(f)
    s = [c for c = s if isdigit(c)]
    sum += parse(Int64, "$(s[1])$(s[end])")
end
close(f)
println(sum)
